lib/framereader.py

```diff
@@ -165,6 +165,26 @@ def read_frame_P210(fd, width, height):
 
   return y, u, v
 
+def read_frame_Y210(fd, width, height):
+  #https://docs.microsoft.com/en-us/windows/win32/medfound/10-bit-and-16-bit-yuv-video-formats#422-formats
+  #Y210 (each pair of pixels is stored as an array of four WORD)
+  #   Y0           U            Y1             V
+  #  word0        word1        word2         word3
+  #bits 15-0     bits15-0     bits15-0      bits15-0
+  #
+  #The 16bit representations described here use little-endian WORD values for each channel
+  #bits:  15 14 13 ...  ...     5 4 3 2 1 0
+  #data:  valid data  ...       0 0 0 0 0 0
+  width2  = (width + 1) // 2
+  size    = width2 * height * 4
+
+  y210 = numpy.fromfile(fd, dtype=numpy.uint16, count=size)
+  y  = y210[0::2].reshape((height, width2*2)) & 0xffc0
+  u  = y210[1::4].reshape((height, width2)) & 0xffc0
+  v  = y210[3::4].reshape((height, width2)) & 0xffc0
```

Comment:
Since this is packed, I think we may have an issue reshaping u and v when we have an odd dimension.  See `read_frame_YUY2` for example.

Reply:
Yes, I also noticed this, so my pre-patch is deal the width as the even alignment. but after compared to the test assert generated by gstreamer, I found that my original deal is not a common method, so I updated the patch to follow the exiting gstreamer deal method.